### PR TITLE
feat!: add new compute tasks status linked to function status

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Companion PR

- https://github.com/Substra/substra-backend/pull/823
- https://github.com/Substra/substra-tests/pull/319
- https://github.com/Substra/substra/pull/397
- https://github.com/Substra/substra-frontend/pull/297
- https://github.com/Substra/substra-documentation/pull/390

## Description

Fixes of FL-1397

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

File: docs/assets/computetask.md [hunk 1]
```diff
@@ -83,20 +86,25 @@ This is to ensure that when a task starts (switch to DOING), all its inputs are
...
-- only the owner can cancel a task
-- only the worker can act on a task processing (DOING/DONE/FAILED)
+- BUILD_STARTED & BUILD_FINISHED are done internally
+- only the owner can cancel a task or act on building (function being built on owner)
+- only the worker can act on a task processing (DOING/DONE)
+- both can fail a task 
 
 ## Worker
 
+A task is built on a worker in the task owner.
```
Comment (SdgJlbl) [inline docs/assets/computetask.md:107]: I don't understand this sentence.

**Review by SdgJlbl — COMMENTED** [review]
Minor comment

File: docs/assets/computetask.md (hunk 1 above)
Comment (guilhem-barthes) [reply docs/assets/computetask.md:107]: Me neither ^^I guess the intent was to say "A function associated with a task is built in the organization owning the function". Is this sentence more clear?

File: docs/assets/schemas/computetask.state.mmd [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+	failed: FAILED
+	done: DONE
+
+
+	[*] --> waitingBuilding
+	[*] --> building
+	[*] --> waitingParent
+	[*] --> waitingExecutor
+	[*] --> failed
+    [*] --> canceled
```
[the remark this answers is not in the document]
Comment (guilhem-barthes) [reply docs/assets/schemas/computetask.state.mmd:17]: Yes, if the associated function is failed or canceled when the task is registered

**Review by SdgJlbl — APPROVED** [review]
Thanks 🙏

File: docs/assets/schemas/computetask.state.mmd (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThibaultFy) [reply docs/assets/schemas/computetask.state.mmd:17]: Ah yes of course, thanks !!

**Pull request merged**